### PR TITLE
Introduce modern unit tests for new test structure

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Introduce new unit tests structure, setup, scripts
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10733
| How to test?  | Travis tests should be green and include the new unit test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11525)

### CODE REVIEW

File: composer.json
Comment (PierreRambaud) [inline composer.json]: PsTests should be a proper name no?

Comment (mickaelandrieu) [reply composer.json]: How about LegacyTests ? :)

Comment (matks) [reply composer.json]: Yes, I had a lack of inspiration 😛

File: tests/check_unit.sh
Comment (mickaelandrieu) [inline tests/check_unit.sh]: PHPUNIT UNIT TESTS :p

**Review by mickaelandrieu — COMMENTED** [review]
Only minor changes, we need to figure out a good name for namespace.

File: tests/README.md
Comment (eternoendless) [inline tests/README.md]: I don't think it's a good idea to say they are maintained by a specific group. Anyone should be able to add/improve E2E tests, starting with the Core team.

```suggestion
- `E2E` and `Selenium` folders contain end-to-end tests
```

File: tests-legacy/README.md
Comment (eternoendless) [inline tests-legacy/README.md]: Small punctuation problem.
```suggestion
used: unit tests with phpunit, unit tests with phpunit using PrestaShop Core classes or a database fixtures,
```

File: tests/README.md
Comment (eternoendless) [inline tests/README.md]: Watch out, "mock" may not the correct term. According to Martin Fowler, there are actually [several kinds of test doubles](https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs):

> Meszaros uses the term Test Double as the generic term for any kind of pretend object used in place of a real object for testing purposes. The name comes from the notion of a Stunt Double in movies. (One of his aims was to avoid using any name that was already widely used.) Meszaros then defined four particular kinds of double:
> 
> - **Dummy** objects are passed around but never actually used. Usually they are just used to fill parameter lists.
> - **Fake objects** actually have working implementations, but usually take some shortcut which makes them not suitable for production (an in memory database is a good example).
> - **Stubs** provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.
> - **Spies** are stubs that also record some information based on how they were called. One form of this might be an email service that records how many messages it was sent.
> - **Mocks** are what we are talking about here: objects pre-programmed with expectations which form a specification of the calls they are expected to receive.

In order to avoid misunderstandings, I suggest this change:
```suggestion
- every dependency of the class must be replaced by [test doubles][1]*

[1]: https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs
```

File: tests/README.md
Comment (eternoendless) [inline tests/README.md]:
```suggestion
- Use camelCase names for test function names.
- Try to make method names explain the *intent* of the test case as best as possible. Don't hesitate to write long method names if necessary.
	- Bad example: `testGetPrice` (no idea what such a test is supposed to do)
	- Good example: `testDiscountIsAppliedToFinalPrice`
```

File: composer.json
Comment (eternoendless) [reply composer.json]: I vote for LegacyTests

**Review by eternoendless — DISMISSED** [review]

File: composer.json
Comment (PierreRambaud) [reply composer.json]: Yep LegacyTests :+1:

**Comment by matks** [comment]
Namespace will be migrated in https://github.com/PrestaShop/PrestaShop/pull/11529
This PR can be merged after.

File: tests/README.md
Comment (jolelievre) [inline tests/README.md]: containS

File: tests/check_unit.sh
Comment (Quetzacoalt91) [inline tests/check_unit.sh]: Why don't you `exit $UNIT` ?

Comment (PierreRambaud) [reply tests/check_unit.sh]: :+1:
```
#!/bin/bash

composer run-script unit-tests --timeout=0;
UNIT=$?

if [[ "$UNIT" == "0" ]]; then
  echo -e "\e[UNIT TESTS OK"
else
  echo -e "\e[91mUNIT TESTS FAILED"
fi
exit $UNIT;
```

**Comment by matks** [comment]
@Quetzacoalt91 @jolelievre Feedbacks implemented in https://github.com/PrestaShop/PrestaShop/pull/11525/commits/7525747cdf88ac7f0d852ea71adf923ff0abba69 😉

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Travis handle the QA step for this one. Merging.

Thank you @matks

**Pull request merged**
